src/main/java/org/prebid/server/bidder/sharethrough/SharethroughBidder.java

```diff
@@ -0,0 +1,200 @@
+    }
+
+    /**
+     * Retrieves from {@link Imp} and filter not valid {@link ExtImpSharethrough} and returns list result with errors.
+     */
+    private List<StrUriParameters> parseBidRequestToUriParameters(BidRequest request) {
+        final List<StrUriParameters> strUriParameters = new ArrayList<>();
+
+        final boolean consentRequired = SharethroughRequestUtil.isConsentRequired(request.getRegs());
+
+        final ExtUser extUser = SharethroughRequestUtil.getExtUser(request.getUser());
+        final String consent = SharethroughRequestUtil.getConsent(extUser);
+
+        final String ua = SharethroughRequestUtil.getUa(request.getDevice());
+        final boolean canAutoPlay = SharethroughRequestUtil.canBrowserAutoPlayVideo(ua);
+
+        for (Imp imp : request.getImp()) {
+            ExtImpSharethrough extImpStr = Json.mapper.<ExtPrebid<?, ExtImpSharethrough>>convertValue(
+                    imp.getExt(), SHARETHROUGH_EXT_TYPE_REFERENCE).getBidder();
+
+            strUriParameters.add(createStrUriParameters(consentRequired, consent, canAutoPlay, imp, extImpStr));
+        }
+        return strUriParameters;
+    }
+
+    /**
+     * Populate {@link StrUriParameters} with publisher request, imp, imp.ext values
+     */
+    private StrUriParameters createStrUriParameters(boolean isConsentRequired, String consentString,
+                                                    boolean canBrowserAutoPlayVideo, Imp imp,
+                                                    ExtImpSharethrough extImpStr) {
+        final Size size = SharethroughRequestUtil.getSize(imp, extImpStr);
+        return StrUriParameters.builder()
+                .pkey(extImpStr.getPkey())
+                .bidID(imp.getId())
+                .consentRequired(isConsentRequired)
+                .consentString(consentString)
+                .instantPlayCapable(canBrowserAutoPlayVideo)
+                .iframe(extImpStr.getIframe())
+                .height(size.getHeight())
+                .width(size.getWidth())
+                .build();
+    }
+
+    /**
+     * Make {@link HttpRequest} from uri and headers
+     */
+    private HttpRequest<Void> makeHttpRequest(String uri, MultiMap headers) {
+        return HttpRequest.<Void>builder()
+                .method(HttpMethod.POST)
+                .uri(uri)
+                .body(null)
+                .headers(headers)
+                .payload(null)
+                .build();
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(HttpCall<Void> httpCall, BidRequest bidRequest) {
+        final HttpResponse httpResponse = httpCall.getResponse();
+
+        final ExtImpSharethroughResponse sharethroughBid;
+        try {
+            sharethroughBid = Json.mapper.readValue(httpResponse.getBody(), ExtImpSharethroughResponse.class);
+            return Result.of(toBidderBid(sharethroughBid, httpCall.getRequest()), Collections.emptyList());
+        } catch (IOException e) {
```

Comment:
Please move the 2nd catch block here to use multi-catch `IOException | IllegalArgumentException e`. We can return `BidderError.badServerResponse` in both cases since we are dealing with a response from their service, i.e. `badServerResponse` is the right one. `badInput` is when we receive a `bidRequest` from the client.

Reply:
In Go they throw BadInput when creative is empty. I change to `badServerResponse`, but be aware